### PR TITLE
Flyout: Update Caret

### PR DESCRIPTION
**Pull request by mminute**
Design has provided a new specification for the Flyout caret.

The caret is no longer used in pinboard except in the case of user education and an additional upcoming use case.
This change implements a smaller caret design.
6xp from base to point.
16px wide.
Rounded point.

## Test Plan
Unit tests updated.
Changes were tested within the Gestalt development environment using sm/md/lg Flyout triggers in order to verify that the caret is positioned as expected.

All

### CODE REVIEW

**Comment by ashleyseto** [comment]
@anniemochi

File: packages/gestalt/src/Caret.js [hunk 1]
```diff
@@ -4,34 +4,38 @@ import PropTypes from 'prop-types';
...
-    <svg width="24" height="24">
+    <svg height={height} width={width}>
       <path d={path} />
     </svg>
   );
 }
 
 Caret.propTypes = {
   direction: PropTypes.oneOf(['up', 'right', 'down', 'left']),
+  height: PropTypes.number,
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Caret.js:39]: nit `PropTypes.oneOf([6, 16]).isRequired`

File: packages/gestalt/src/Contents.js [hunk 2]
```diff
@@ -204,22 +205,23 @@ export function calcEdgeShifts(
...
-  const isCloseHorizontally =
-    triggerRect.left - flyoutHorizontalShift < 0 ||
-    triggerRect.right + flyoutHorizontalShift > windowSize.width;
   if (isCloseVertically) {
     flyoutVerticalShift =
       BORDER_RADIUS - (triggerRect.height - CARET_HEIGHT) / 2;
     caretVerticalShift = BORDER_RADIUS;
   }
+
+  const isCloseHorizontally =
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Contents.js:222]: just moved from 215 line

File: packages/gestalt/src/Contents.js [hunk 3]
```diff
@@ -24,10 +24,11 @@ const DIR_INDEX_MAP = {
...
 };
 
-const MARGIN = 24;
-export const CARET_HEIGHT = 24;
-const CARET_OFFSET_FROM_SIDE = 24;
 export const BORDER_RADIUS = 8;
+export const CARET_HEIGHT = 6;
+const CARET_OFFSET_FROM_SIDE = 24;
+export const CARET_WIDTH = 16;
+const MARGIN = 24;
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Contents.js:31]: alpha and new valuees

**Comment by AlbertCarreras** [comment]
LGTM

**Comment by anniemochi** [comment]
Could we use a shadow instead of a border? Also, it looks a bit squashed. Happy to jam with you offline @mminute

**Comment by mminute** [comment]
Updated for more rounded carets
RoundedCaret

File: packages/gestalt/src/Caret.js [hunk 4]
```diff
@@ -2,36 +2,37 @@
 import React, { type Node } from 'react';
 import PropTypes from 'prop-types';
 
+const PATHS = {
+  down:
+    'M 0 4 C 0.69415 4 1.35929 3.7216 1.84646 3.2271 L 4.3752 0.66038 C 5.182 -0.15852 6.5628 -0.224911 7.4592 0.512109 C 7.5161 0.558899 7.5703 0.60839 7.6215 0.66038 L 10.148 3.2248 C 10.6367 3.7208 11.3038 4 12 4',
```
Comment (christianvuerings) [inline packages/gestalt/src/Caret.js:7]: @mminute would it be possible to run the original SVG through `svgo`? https://github.com/svg/svgo

That way we can lower the file size of these inline SVGs.

**Pull request merged**